src/opensearchDatasource.ts

```diff
@@ -302,78 +328,83 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
+      if (!fieldName) {
+        return;
+      }
 
-        const fieldNames = fieldName.split('.');
-        let fieldValue = source;
+      const fieldNames = fieldName.split('.');
+      let fieldValue = source;
 
-        for (let i = 0; i < fieldNames.length; i++) {
-          fieldValue = fieldValue[fieldNames[i]];
-          if (!fieldValue) {
-            console.log('could not find field in annotation: ', fieldName);
-            return '';
-          }
+      for (let i = 0; i < fieldNames.length; i++) {
+        fieldValue = fieldValue[fieldNames[i]];
+        if (!fieldValue) {
+          console.log('could not find field in annotation: ', fieldName);
+          return '';
         }
+      }
 
-        return fieldValue;
-      };
+      return fieldValue;
+    };
 
-      for (let i = 0; i < hits.length; i++) {
-        const source = hits[i]._source;
-        let time = getFieldFromSource(source, timeField);
-        if (typeof hits[i].fields !== 'undefined') {
-          const fields = hits[i].fields;
-          if (_.isString(fields[timeField]) || _.isNumber(fields[timeField])) {
-            time = fields[timeField];
-          }
+    for (let i = 0; i < hits.length; i++) {
+      const source = hits[i]._source;
+      let time = getFieldFromSource(source, timeField);
+      if (typeof hits[i].fields !== 'undefined') {
+        const fields = hits[i].fields;
+        if (_.isString(fields[timeField]) || _.isNumber(fields[timeField])) {
+          time = fields[timeField];
         }
+      }
 
-        const event: {
-          annotation: any;
-          time: number;
-          timeEnd?: number;
-          text: string;
-          tags: string | string[];
-        } = {
-          annotation: annotation,
-          time: toUtc(time).valueOf(),
-          text: getFieldFromSource(source, textField),
-          tags: getFieldFromSource(source, tagsField),
-        };
-
-        if (timeEndField) {
-          const timeEnd = getFieldFromSource(source, timeEndField);
-          if (timeEnd) {
-            event.timeEnd = toUtc(timeEnd).valueOf();
-          }
-        }
+      const event: {
+        annotation: any;
+        time: number;
+        timeEnd?: number;
+        text: string;
+        tags?: string[];
+      } = {
+        annotation: annotation,
+        time: toUtc(time).valueOf(),
+        text: getFieldFromSource(source, textField),
+      };
 
-        // legacy support for title field
-        if (annotation.titleField) {
-          const title = getFieldFromSource(source, annotation.titleField);
-          if (title) {
-            event.text = title + '\n' + event.text;
-          }
+      if (timeEndField) {
+        const timeEnd = getFieldFromSource(source, timeEndField);
+        if (timeEnd) {
+          event.timeEnd = toUtc(timeEnd).valueOf();
         }
+      }
 
-        if (typeof event.tags === 'string') {
-          event.tags = event.tags.split(',');
+      // legacy support for title field
```

Comment:
Do we know what this title field is, was it ever supported in OpenSearch?

Reply:
It looks like this was added in the first commit, so possibly not, but I'm going to leave it for now just in case.